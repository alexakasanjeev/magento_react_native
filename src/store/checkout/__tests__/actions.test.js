import {
  getCountries,
  placeCartOrder,
  getOrderDetail,
  resetPaymentState,
  getShippingMethod,
  resetCheckoutState,
  resetShippingState,
  addCartShippingInfo,
  addCartBillingAddress,
  resetCheckoutAddressState,
} from '../actions';

describe('checkout actions', () => {
  test('getCountries()', () => {
    // Exercise
    const result = getCountries();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('addCartBillingAddress()', () => {
    // Setup
    const input = {
      address: {
        region: 'California',
        region_id: 2,
        region_code: 'CA',
        country_id: 'US',
        street: ['Elm street'],
        telephone: 123456789,
        postcode: 123456,
        city: 'west world',
        firstname: 'Sanjeev',
        lastname: 'Yadav',
        same_as_billing: 1,
      },
      useForShipping: true,
    };

    // Exercise
    const result = addCartBillingAddress(input);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getShippingMethod()', () => {
    // Setup
    const input = {
      address: {
        region: 'California',
        region_id: 2,
        region_code: 'CA',
        country_id: 'US',
        street: ['Elm street'],
        telephone: 123456789,
        postcode: 123456,
        city: 'West world',
        firstname: 'Sanjeev',
        lastname: 'Yadav',
      },
    };

    // Exercise
    const result = getShippingMethod(input);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('addCartShippingInfo()', () => {
    // Setup
    const input = {
      addressInformation: {
        shipping_address: {
          region: 'California',
          region_id: 2,
          region_code: 'CA',
          country_id: 'US',
          street: 'Elm Street',
          telephone: 123456789,
          postcode: 123456,
          city: 'West world',
          firstname: 'Sanjeev',
          lastname: 'Yadav',
          email: 'sanjeev@gmail.com',
        },
        billing_address: {
          region: 'California',
          region_id: 2,
          region_code: 'CA',
          country_id: 'US',
          street: 'Elm Street',
          telephone: 123456789,
          postcode: 123456,
          city: 'West world',
          firstname: 'Sanjeev',
          lastname: 'Yadav',
          email: 'sanjeev@gmail.com',
        },
        shipping_method_code: 'FREE',
        shipping_carrier_code: 'FREE',
        extension_attributes: {},
      }
    };

    // Exercise
    const result = addCartShippingInfo(input);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('placeCartOrder()', () => {
    // Setup
    const paymentInformation = {
      billingAddress: {
        city: 'West world',
        country_id: 'US',
        email: 'sanjeev@gmail.com',
        firstname: 'Sanjeev',
        lastname: 'Yadav',
        postcode: '123456',
        region: 'California',
        region_code: 'CA',
        region_id: 2,
        street: ['Elm Street'],
        telephone: '123456789',
      },
      paymentMethod: {
        method: 'checkmo'
      }
    };

    // Exercise
    const result = placeCartOrder(paymentInformation);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getOrderDetail()', () => {
    // Setup
    const orderId = 33;

    // Exercise
    const result = getOrderDetail(orderId);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetCheckoutState()', () => {
    // Exercise
    const result = resetCheckoutState();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetCheckoutAddressState()', () => {
    // Exercise
    const result = resetCheckoutAddressState();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetShippingState()', () => {
    // Exercise
    const result = resetShippingState();

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetPaymentState()', () => {
    // Exercise
    const result = resetPaymentState();

    // Verify
    expect(result).toMatchSnapshot();
  });
});
