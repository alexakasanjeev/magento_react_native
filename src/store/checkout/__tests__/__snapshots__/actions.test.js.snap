// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout actions addCartBillingAddress() 1`] = `
Object {
  "payload": Object {
    "address": Object {
      "address": Object {
        "city": "west world",
        "country_id": "US",
        "firstname": "Sanjeev",
        "lastname": "Yadav",
        "postcode": 123456,
        "region": "California",
        "region_code": "CA",
        "region_id": 2,
        "same_as_billing": 1,
        "street": Array [
          "Elm street",
        ],
        "telephone": 123456789,
      },
      "useForShipping": true,
    },
  },
  "type": "MAGENTO_ADD_CART_BILLING_ADDRESS_REQUEST",
}
`;

exports[`checkout actions addCartShippingInfo() 1`] = `
Object {
  "payload": Object {
    "address": Object {
      "addressInformation": Object {
        "billing_address": Object {
          "city": "West world",
          "country_id": "US",
          "email": "sanjeev@gmail.com",
          "firstname": "Sanjeev",
          "lastname": "Yadav",
          "postcode": 123456,
          "region": "California",
          "region_code": "CA",
          "region_id": 2,
          "street": "Elm Street",
          "telephone": 123456789,
        },
        "extension_attributes": Object {},
        "shipping_address": Object {
          "city": "West world",
          "country_id": "US",
          "email": "sanjeev@gmail.com",
          "firstname": "Sanjeev",
          "lastname": "Yadav",
          "postcode": 123456,
          "region": "California",
          "region_code": "CA",
          "region_id": 2,
          "street": "Elm Street",
          "telephone": 123456789,
        },
        "shipping_carrier_code": "FREE",
        "shipping_method_code": "FREE",
      },
    },
  },
  "type": "MAGENTO_ADD_CART_SHIPPING_INFO_REQUEST",
}
`;

exports[`checkout actions getCountries() 1`] = `
Object {
  "type": "MAGENTO_COUNTRIES_REQUEST",
}
`;

exports[`checkout actions getOrderDetail() 1`] = `
Object {
  "payload": Object {
    "orderId": 33,
  },
  "type": "MAGENTO_ORDER_DETAIL_REQUEST",
}
`;

exports[`checkout actions getShippingMethod() 1`] = `
Object {
  "payload": Object {
    "address": Object {
      "address": Object {
        "city": "West world",
        "country_id": "US",
        "firstname": "Sanjeev",
        "lastname": "Yadav",
        "postcode": 123456,
        "region": "California",
        "region_code": "CA",
        "region_id": 2,
        "street": Array [
          "Elm street",
        ],
        "telephone": 123456789,
      },
    },
  },
  "type": "MAGENTO_GET_SHIPPING_METHOD_REQUEST",
}
`;

exports[`checkout actions placeCartOrder() 1`] = `
Object {
  "payload": Object {
    "paymentInformation": Object {
      "billingAddress": Object {
        "city": "West world",
        "country_id": "US",
        "email": "sanjeev@gmail.com",
        "firstname": "Sanjeev",
        "lastname": "Yadav",
        "postcode": "123456",
        "region": "California",
        "region_code": "CA",
        "region_id": 2,
        "street": Array [
          "Elm Street",
        ],
        "telephone": "123456789",
      },
      "paymentMethod": Object {
        "method": "checkmo",
      },
    },
  },
  "type": "MAGENTO_PLACE_CART_ORDER_REQUEST",
}
`;

exports[`checkout actions resetCheckoutAddressState() 1`] = `
Object {
  "type": "RESET_CHECKOUT_ADDRESS_STATE",
}
`;

exports[`checkout actions resetCheckoutState() 1`] = `
Object {
  "type": "RESET_CHECKOUT_STATE",
}
`;

exports[`checkout actions resetPaymentState() 1`] = `
Object {
  "type": "RESET_PAYMENT_STATE",
}
`;

exports[`checkout actions resetShippingState() 1`] = `
Object {
  "type": "RESET_SHIPPING_STATE",
}
`;
